packages/react-scripts/scripts/build.js

```diff
@@ -57,6 +57,31 @@ measureFileSizesBeforeBuild(paths.appBuild)
     fs.emptyDirSync(paths.appBuild);
     // Merge with the public folder
     copyPublicFolder();
+
+    // Check if every vendors are defined in the package.json
+    // @remove-on-eject-begin
+    // The devDepencencies shouldn't be available for vendors
+    // But otherwise the tests fail.
+    // @remove-on-eject-end
+    const packageJson = require(paths.appPackageJson);
+    const dependencies = Object.keys(packageJson.dependencies || {})
+      .concat(Object.keys(packageJson.devDependencies || {}))
+      .concat(Object.keys(packageJson.peerDependencies || {}));
+    const vendors = fs.existsSync(paths.appVendors)
+      ? require(paths.appVendors)
+      : [];
+    const missingVendors = vendors.filter(
+      vendor => dependencies.indexOf(vendor) === -1
+    );
+    if (missingVendors.length > 0) {
+      throw new Error(
+        'Error: Unknown vendors: ' +
+          chalk.yellow(missingVendors) +
+          " should be listed in the project's dependencies.\n" +
+          `(Vendors defined in '${path.resolve(paths.appVendors)}')`
+      );
+    }
```

Comment:
This is quite a bit of code. I don't think we want ejected users to end up with this. Let's move this somewhere into `react-dev-utils`?

Reply:
in a `detectMissingVendors` function ?

Reply:
Something like this, maybe.

Reply:
What about adding the ESLint rule [`import/no-extraneous-dependencies`](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-extraneous-dependencies.md) to eslint-config-react-app instead.